### PR TITLE
feat(imageBrush): Enhance ImageBrush support

### PR DESCRIPTION
**Pull request by dr1rrb**
fixes https://github.com/unoplatform/uno/issues/101

## Feature
* iOS / macOS / Android: `ImageBrush` can now handle `WriteableBitmap`
* WASM: `ImageBrush` is now supported

## What is the current behavior?
* iOS / macOS / Android: If a `WriteableBitmap` is set a source of an `ImageBrush`, the brush fails silently
* WASM: Settings an `ImageBrush` as `Background` of any element is silently ignored

## What is the new behavior?
🙃

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Media/ImageBrushTests/BorderImageBrush.xaml.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 namespace Uno.UI.Samples.UITests.ImageBrushTestControl
 {
-	[SampleControlInfo("ImageBrushTestControl", "BorderImageBrush")]
+	[SampleControlInfo("ImageBrush", "BorderImageBrush")]
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Media/ImageBrushTests/BorderImageBrush.xaml.cs:19]: You can't validate for visual breaking changes if you change the category name.

File: src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs [hunk 2]
```diff
@@ -347,18 +347,14 @@ private void TryOpenImage()
...
-						SetFromWriteableBitmap(wb);
-					}
 					// We want to reset the image when there is no source provided.
-					else if (!imageSource?.HasSource() ?? true)
+					if (!imageSource?.HasSource() ?? true)
 					{
 						ResetSource();
 					}
-					else if (imageSource.ImageData != null)
+					else if (imageSource.TryOpenSync(out var bitmap))
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs:355]: I agree with the robot :)

File: src/Uno.UI/UI/Xaml/Media/ImageSource.wasm.cs [hunk 3]
```diff
@@ -20,11 +24,201 @@
 
 namespace Windows.UI.Xaml.Media
 {
-	partial class ImageSource 
+	internal enum ImageDataKind
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Media/ImageSource.wasm.cs:27]: Other file

File: src/Uno.UI/UI/Xaml/Media/ImageSource.wasm.cs [hunk 4]
```diff
@@ -20,11 +24,201 @@
...
+		/// <summary>
+		/// The image failed to load (cf. The Error property)
+		/// </summary>
+		Error = 256
+	}
+
+	/// <summary>
+	/// Represents the raw data of an **opened** image source
+	/// </summary>
+	internal struct ImageData
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Media/ImageSource.wasm.cs:50]: Other file

File: src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/Image/Image.Android.cs:355]: No, if `imageSource` is `null` we will go in the first case and do `ResetSource()`

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Media/ImageBrushTests/BorderImageBrush.xaml.cs (hunk 1 above)
Comment (dr1rrb) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Media/ImageBrushTests/BorderImageBrush.xaml.cs:19]: Right, I can undo this rename, and change it later in a separated commit so we will be sure to not have any bug in it.

**Pull request merged**